File: sdk/csharp/libraries/microsoft.bot.solutions/Dialogs/EventPrompt.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;
using System.Threading;
using System.Threading.Tasks;
using Microsoft.Bot.Builder;
using Microsoft.Bot.Builder.Dialogs;
using Microsoft.Bot.Schema;

namespace Microsoft.Bot.Solutions
{
    /// <summary>
    /// Event prompt that enables Bots to wait for a incoming event matching a given name to be received.
    /// </summary>
    [ExcludeFromCodeCoverageAttribute]
    [Obsolete("This class is being deprecated. For more information, refer to https://aka.ms/bfvarouting.", false)]
    public class EventPrompt : ActivityPrompt
    {
        public EventPrompt(string dialogId, string eventName, PromptValidator<Activity> validator)
            : base(dialogId, validator)
        {
            EventName = eventName ?? throw new ArgumentNullException(nameof(eventName));
        }

        public string EventName { get; set; }

        protected override Task<PromptRecognizerResult<Activity>> OnRecognizeAsync(ITurnContext turnContext, IDictionary<string, object> state, PromptOptions options, CancellationToken cancellationToken)
        {
            var result = new PromptRecognizerResult<Activity>();
            var activity = turnContext.Activity;

            if (activity.Type == ActivityTypes.Event && !string.IsNullOrEmpty(activity.Name))
            {
                var ev = activity.AsEventActivity();

                if (ev.Name == EventName)
                {
                    result.Succeeded = true;
                    result.Value = turnContext.Activity;
                }
            }

            return Task.FromResult(result);
        }
    }
}

File: sdk/typescript/libraries/bot-solutions/src/dialogs/eventPrompt.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { TurnContext } from 'botbuilder';
import { ActivityPrompt, PromptOptions, PromptRecognizerResult, PromptValidator } from 'botbuilder-dialogs';
import { Activity, ActivityTypes, IEventActivity } from 'botframework-schema';

/**
 * Event prompt that enables Bots to wait for a incoming event matching a given name to be received.
 */
//OBSOLETE: This class is being deprecated. For more information, refer to https://aka.ms/bfvarouting.
export class EventPrompt extends ActivityPrompt {
    public eventName: string;

    public constructor(dialogId: string, eventName: string, validator: PromptValidator<Activity>) {
        super(dialogId, validator);

        if (eventName === undefined) { throw new Error ('The value of eventName is undefined'); }
        this.eventName = eventName;
    }

    protected async onRecognize(context: TurnContext, state: object, options: PromptOptions): Promise<PromptRecognizerResult<Activity>> {
        const result: PromptRecognizerResult<Activity> = { succeeded: false };
        const activity: Activity = context.activity;

        if (activity.type === ActivityTypes.Event && activity.name !== undefined && activity.name.trim().length > 0) {
            const ev: IEventActivity = activity as IEventActivity;

            if(ev.name === this.eventName) {
                result.succeeded = true;
                result.value = context.activity;
            }
        }

        return Promise.resolve(result);
    }
}
